### PR TITLE
[8.0][ADD] product_variant_sale_price module

### PR DESCRIPTION
**Pull request by tafaRU**

### CODE REVIEW

File: product_variant_sale_price/post_init.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+from openerp import SUPERUSER_ID
+
+
+def post_init_hook(cr, registry):
+    """
+    Post-install script. Because 'lst_price' has been replaced by
+    'variant_lst_price, we should be able to retrieve its values
+    and use them to fill the new fields.
+    """
+    product_obj = registry['product.product']
```
Comment (pedrobaeza) [inline product_variant_sale_price/post_init.py:14]: Make only one query:

```
UPDATE product_product
SET variant_lst_price = product_template.list_price
FROM product_template
WHERE product_template.id = product_product.product_tmpl_id
```

**Comment by pedrobaeza** [comment]
Isn't maybe better to make the field `list_price` a regular field and remove the standard inverse computation?

**Comment by eLBati** [comment]
@pedrobaeza hi.
We chose to not touch `lst_price` field in order to not remove the standard behaviour: if in some cases the standard behaviour is fine, user can use it, otherwise they use the new field `variant_lst_price`

**Comment by pedrobaeza** [comment]
That's why I prefer to change it, so that you don't have to touch every process that deals with this field.

**Comment by eLBati** [comment]
I mean: with the current version, if users want , for some products, to express prices based on standard `lst_price`, they can use the standard pricelist. If they want some other products to be based on `Variant Sale Price`, they can use `Variant Sale Pricelist`

Could you describe an example where you would have to touch other processes that deals with this field?

Thanks

**Comment by pedrobaeza** [comment]
In a quick search, here you have a hardcoded reference to that field that is not covered by pricelists: https://github.com/odoo/odoo/blob/8.0/addons/account/account_invoice.py#L1359

**Comment by eLBati** [comment]
Ok, I agree, better to touch `lst_price` trying as you suggested

File: product_variant_sale_price/i18n/product_variant_sale_price.pot [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Translation of Odoo Server.
```
Comment (pedrobaeza) [inline product_variant_sale_price/i18n/product_variant_sale_price.pot:1]: Remove POT file. It's not needed anymore.

File: product_variant_sale_price/models/product_product.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp.osv import orm, fields
+from openerp.addons import decimal_precision as dp
+
+
+class ProductProduct(orm.Model):
+    _inherit = 'product.product'
+
+    _columns = {
```
Comment (pedrobaeza) [inline product_variant_sale_price/models/product_product.py:12]: Can you put this on new API?

**Comment by pedrobaeza** [comment]
The field still remains with the old name.

**Comment by pedrobaeza** [comment]
See alternative module proposition here (still a WIP): https://github.com/sergio-teruel/product-variant/blob/8.0-PR-product_variant_sale_price/product_variant_sale_price

**Comment by eLBati** [comment]
Hi @pedrobaeza and @sergio-teruel  know what is the status of the module
https://github.com/sergio-teruel/product-variant/tree/8.0-PR-product_variant_sale_price/product_variant_sale_price
?

Can a PR be done?

**Comment by pedrobaeza** [comment]
Yes, we'll do it next week and work over that one.

**Comment by hurrinico** [comment]
@pedrobaeza i've tried to install sergio's module but it returns this error:
loading product_variant_sale_price/views/product_view.xml
Error context:
View `product.product.form.inherited`
[view_id: 1437, xml_id: product_variant_sale_price.product_normal_form_view_default_check, model: product.product, parent_id: 258]

**Comment by hurrinico** [comment]
@pedrobaeza, if it's ok to you, i'm going to work on the error tomorrow.

**Comment by pedrobaeza** [comment]
OK, sorry for the silence. We have been very busy these days. I'll review it when you have it.

**Comment by hurrinico** [comment]
@pedrobaeza i've done this https://github.com/OCA/product-variant/pull/26 based on @sergio-teruel 's module

**Comment by tafaRU** [comment]
@sergio-teruel @pedrobaeza I'm going to port https://github.com/sergio-teruel/product-variant/tree/8.0-PR-product_variant_sale_price/product_variant_sale_price to OCA and to close the current PR.
Meanwhile, could you please update me about its status?

Thank you in advance.

**Comment by tafaRU** [comment]
I close this in favour of https://github.com/OCA/product-variant/pull/30

**Pull request closed without merging**